mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -320,37 +324,45 @@ private boolean updateFlags(MailboxMessage message, long oldModSeq) {
                 .modSeq(message.getModSeq())
                 .flags(message.createFlags())
                 .build();
-        CompletableFuture<Boolean> messageIdFuture = messageIdDAO.updateMetadata(composedMessageIdWithMetaData, oldModSeq);
-        CompletableFuture<Boolean> imapUidFuture = imapUidDAO.updateMetadata(composedMessageIdWithMetaData, oldModSeq);
-        return messageIdFuture.join() && imapUidFuture.join();
+        return imapUidDAO.updateMetadata(composedMessageIdWithMetaData, oldModSeq)
+            .thenCompose(success -> {
```

Comment:
we usually rely on optional monad to avoid in this case, am I missing something ?